### PR TITLE
nixos/mediawiki: make apache2 optional

### PR DESCRIPTION
**Pull request by Mic92**
###### Description of changes

The next step would be to support nginx.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-apps/mediawiki.nix [hunk 1]
```diff
@@ -212,6 +220,13 @@ in
         example = "/run/keys/mediawiki-password";
       };
 
+      passwordSender = mkOption {
+        type = types.str;
+        default = if cfg.webserver == "httpd" then if cfg.httpd.virtualHost.adminAddr != null then cfg.httpd.virtualHost.adminAddr else config.services.httpd.adminAddr else "root@localhost";
+        defaultText = literalExpression ''if cfg.webserver == "httpd" then if cfg.httpd.virtualHost.adminAddr != null then cfg.httpd.virtualHost.adminAddr else config.services.httpd.adminAddr else "root@localhost"'';
```
Comment (mweinelt) [inline nixos/modules/services/web-apps/mediawiki.nix:226]:
```suggestion
        default = 
          if cfg.webserver == "httpd" then
             if cfg.httpd.virtualHost.adminAddr != null then
               cfg.httpd.virtualHost.adminAddr
             else
               config.services.httpd.adminAddr
           else "root@localhost";
        defaultText = literalExpression ''
          if cfg.webserver == "httpd" then
            if cfg.httpd.virtualHost.adminAddr != null
              then
                cfg.httpd.virtualHost.adminAddr
              else
                config.services.httpd.adminAddr
            else
              "root@localhost"'';
```

File: nixos/tests/mediawiki.nix [hunk 2]
```diff
@@ -54,4 +54,15 @@ in
...
   };
+
+  nohttpd = testLib.makeTest {
+    name = "mediawiki-nohttpd";
+    nodes.machine = {
+      services.mediawiki.webserver = "none";
+    };
+    testScript = ''
+      start_all()
+      machine.wait_for_unit("phpfpm-mediawiki.service")
```
Comment (mweinelt) [inline nixos/tests/mediawiki.nix:65]: Would be good to put this behind a webserver to test if  we get a HTTP/200.

Comment (Mic92) [reply nixos/tests/mediawiki.nix:65]: I tested the fastcgi socket instead.

File: nixos/modules/services/web-apps/mediawiki.nix [hunk 3]
```diff
@@ -241,6 +285,12 @@ in
         '';
       };
 
+      webserver = mkOption {
+        type = types.enum [ "apache" "none" ];
```
Comment (mweinelt) [inline nixos/modules/services/web-apps/mediawiki.nix:289]: Inconsistent use of `apache` and `httpd` throughout the module.

Comment (Mic92) [reply nixos/modules/services/web-apps/mediawiki.nix:289]: Well what do you prefer?For the nixos options we use httpd. However when specifiing a Webserver name it's httpd

Comment (mweinelt) [reply nixos/modules/services/web-apps/mediawiki.nix:289]: What I meant is that you currently doing the following matches:

- 4x `if (cfg.webserver == "httpd")`
- 5x `if (cfg.webserver == "apache")`

The `httpd` case should probably mean `apache` iiuc.

**Comment by mweinelt** [comment]
@ofborg tet mediawiki

File: nixos/modules/services/web-apps/mediawiki.nix [hunk 4]
```diff
@@ -8,7 +8,8 @@ let
   cfg = config.services.mediawiki;
   fpm = config.services.phpfpm.pools.mediawiki;
   user = "mediawiki";
-  group = config.services.httpd.group;
+  group = if cfg.webserver == "apache" then "apache" else "mediawiki";
```
[the remark this answers is not in the document]
Comment (nomeata) [reply nixos/modules/services/web-apps/mediawiki.nix:11]: Moved to https://github.com/NixOS/nixpkgs/issues/236446

**Pull request merged**